src/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import { OctokitOptions } from "@octokit/core/dist-types/types.d";
+
+type LimitHandler = (retryAfter: number, options: any, octokit: any) => void;
+
+export interface ThrottlingOptions extends OctokitOptions {
+  throttle: {
+    enabled?: boolean;
+    Bottleneck?: any;
+    id?: string;
+    timeout?: number;
+    connection?: any;
+    onAbuseLimit: LimitHandler;
+    onRateLimit: LimitHandler;
+  };
+}
+
+export type OctokitThrottlingOptions = OctokitOptions | ThrottlingOptions;
```

Comment:
A loooong time without touching TS. If I don't do this `Union` the compiler does not allow me to use `ThrottlingOptions` in a place where `OctokitOptions` is expected.

Doing this union we are not making `options.throttle` mandatory which is not good.

I still have to play around, maybe with generics `OctokitOptions<T>`?

Reply:
If I understand the code properly, it would be sufficient to use `ThrottlingOptions` directly instead of `OctokitThrottlingOptions`, and then we should rename it to `OctokitThrottlingOptions`